File: articles/judgment-under-uncertainty/r/run_all_judgment_workflows.R
# run_all_judgment_workflows.R
# Run all R workflows for Judgment Under Uncertainty.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "judgment_under_uncertainty_workflow.R",
  "calibration_review_tables.R",
  "confidence_error_reports.R",
  "bayesian_revision_profiles.R",
  "evidence_quality_diagnostics.R",
  "judgment_review_tables.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R judgment workflows completed.")

File: articles/judgment-under-uncertainty/python/run_all_judgment_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Judgment Under Uncertainty."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "bayesian_update_diagnostics.py",
    "calibration_error_scoring.py",
    "confidence_gap_analysis.py",
    "anchoring_distortion_model.py",
    "evidence_quality_profiles.py",
    "judgment_review_queue.py",
    "decision_record_exporter.py",
    "judgment_under_uncertainty_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python judgment workflows completed.")


if __name__ == "__main__":
    main()
